### PR TITLE
Update hub cli commands to use the latest API

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

- Initially we were fetching the resource yamls from github,
now since we have `/readme` and `/yaml` endpoints we get
the data from api itself, so this patch calls the `/yaml`
api to fetch the resource yaml

- With this patch we are updating the following commands
    - get
    - install
    - reinstall
    - upgrade
    - downgrade

Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [ ] Run UI Unit Tests, Lint Checks with `make ui-check`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

File: api/test/e2e/get/get_test.go [hunk 1]
```diff
@@ -16,45 +16,33 @@
...
 
-	"github.com/AlecAivazis/survey/v2/terminal"
-	"github.com/Netflix/go-expect"
 	"github.com/stretchr/testify/assert"
 	"github.com/tektoncd/hub/api/test/cli"
 	"gotest.tools/v3/icmd"
 )
 
-const expected = `---
+var task2 = `---
```
Comment (vinamra28) [inline api/test/e2e/get/get_test.go:28]: can we rename it to some other name, maybe `expected` ?

File: api/test/e2e/get/get_test.go [hunk 2]
```diff
@@ -16,45 +16,33 @@
...
-    description: tkn CLI arguments to run
-  steps:
-    - name: tkn
-      image: "$(params.tkn-image)"
-      command: ["/usr/local/bin/tkn"]
-      args: ["$(params.ARGS)"]
-
+    v0.1 Task to run foo
 `
+var yamlJson = []byte(`{
```
Comment (vinamra28) [inline api/test/e2e/get/get_test.go:41]: same here. How about `apiResponse` ?

File: api/pkg/cli/cmd/downgrade/downgrade.go [hunk 3]
```diff
@@ -150,15 +151,27 @@ func (opts *options) run() error {
...
+
+	manifest, err := opts.hubRes.ResourceYaml()
 	if err != nil {
 		return err
 	}
 
+	// manifest, err := opts.hubResVerResult.VersionManifest(opts.version)
+	// if err != nil {
+	// 	return err
+	// }
```
Comment (vinamra28) [inline api/pkg/cli/cmd/downgrade/downgrade.go:169]: can we please remove this?

**Comment by pratap0007** [comment]
lgtm

File: api/test/e2e/get/get_test.go [hunk 4]
```diff
@@ -70,35 +58,18 @@ func TestGetIneractiveE2E(t *testing.T) {
...
-				if _, err := c.ExpectEOF(); err != nil {
-					return err
-				}
+	t.Run("Result for get command when resource name and version is passed", func(t *testing.T) {
 
-				c.Close()
-				return nil
-			}})
-	})
+		go func() {
```
Comment (sm43) [inline api/test/e2e/get/get_test.go:64]: would be better if we `defer` the `shutdown` for the http server?

**Pull request merged**